### PR TITLE
Support BuildKit

### LINKED ISSUE #187: Rootful BuildKit + SELinux:  Dockerfile `RUN` command throws error code 139
When SELinux is enforcing and BuildKit is running as Rootful mode, every Dockerfile `RUN` command throws error code 139.
Rootless mode seems unaffected.

```console
# getenforce 
Enforcing

# cat Dockerfile 
FROM alpine
RUN ls

# buildctl build --frontend dockerfile.v0 --local dockerfile=. --local context=.
[+] Building 2.9s (5/5) FINISHED                                                                                                                                                                                                                                         
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                0.1s
 => => transferring dockerfile: 56B                                                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                   0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                                                    2.7s
 => CACHED [1/2] FROM docker.io/library/alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad                                                                                                                                              0.0s
 => => resolve docker.io/library/alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad                                                                                                                                                     0.0s
 => ERROR [2/2] RUN ls                                                                                                                                                                                                                                              0.1s
------
 > [2/2] RUN ls:
------
Dockerfile:2
--------------------
   1 |     FROM alpine
   2 | >>> RUN ls
   3 |     
--------------------
error: failed to solve: process "/bin/sh -c ls" did not complete successfully: exit code: 139
```


Daemon logs:
```
# buildkitd
INFO[2022-10-18T14:05:52Z] auto snapshotter: using overlayfs            
WARN[2022-10-18T14:05:52Z] using host network as the default            
WARN[2022-10-18T14:05:52Z] git source cannot be enabled: failed to find git binary: exec: "git": executable file not found in $PATH 
INFO[2022-10-18T14:05:52Z] found worker "3to1fo25rmz1hwacdyyyu5kog", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:lima-fedora org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.oci.process-mode:sandbox org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/amd64/v2 linux/amd64/v3 linux/amd64/v4 linux/386] 
WARN[2022-10-18T14:05:52Z] using host network as the default            
WARN[2022-10-18T14:05:52Z] git source cannot be enabled: failed to find git binary: exec: "git": executable file not found in $PATH 
INFO[2022-10-18T14:05:52Z] found worker "k8jntgsnmdl8m2e15pqv60sem", labels=map[org.mobyproject.buildkit.worker.containerd.namespace:buildkit org.mobyproject.buildkit.worker.containerd.uuid:4aa6134d-7969-409c-a3b4-8216a502d73b org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:lima-fedora org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/amd64/v2 linux/amd64/v3 linux/amd64/v4 linux/386] 
INFO[2022-10-18T14:05:52Z] found 2 workers, default="3to1fo25rmz1hwacdyyyu5kog" 
WARN[2022-10-18T14:05:52Z] currently, only the default worker can be used. 
INFO[2022-10-18T14:05:52Z] running server on /run/buildkit/buildkitd.sock 
INFO[2022-10-18T14:06:06Z] detected 127.0.0.53 nameserver, assuming systemd-resolved, so using resolv.conf: /run/systemd/resolve/resolv.conf 
ERRO[2022-10-18T14:06:07Z] /moby.buildkit.v1.Control/Solve returned error: rpc error: code = Unknown desc = process "/bin/sh -c ls" did not complete successfully: exit code: 139
```

Version info:
```console
# uname -a
Linux lima-fedora 5.19.15-201.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Oct 13 18:58:38 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

# rpm -qa | grep -i selinux | sort
container-selinux-2.190.0-1.fc36.noarch
libselinux-3.3-4.fc36.x86_64
libselinux-utils-3.3-4.fc36.x86_64
python3-libselinux-3.3-4.fc36.x86_64
rpm-plugin-selinux-4.17.1-3.fc36.x86_64
selinux-policy-36.16-1.fc36.noarch
selinux-policy-targeted-36.16-1.fc36.noarch

# buildkitd --version
buildkitd github.com/moby/buildkit v0.10.4 a2ba6869363812a210fcc3ded6926757ab780b5f

# runc --version
runc version 1.1.4
commit: v1.1.4-0-g5fd4c4d1
spec: 1.0.2-dev
go: go1.19.1
libseccomp: 2.5.1
```

Originally reported by @pcramasamy in
- https://github.com/containerd/nerdctl/issues/1413

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Tested with BuildKit v0.10.5 on Fedora 36.

Fix #187

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@AkihiroSuda BuildKitD is launching containers correct for RUN option.

**Comment by rhatdan** [comment]
@AkihiroSuda Did this change fix the problem you were seeing on https://github.com/containers/container-selinux/issues/187?

**Comment by AkihiroSuda** [comment]
> @AkihiroSuda Did this change fix the problem you were seeing on #187?

Yes

**Comment by AkihiroSuda** [comment]
`RUN ls` works, but `RUN apk add neofetch` does not 😞 

```console
# buildctl build --frontend dockerfile.v0 --local dockerfile=. --local context=.
[+] Building 3.3s (5/5) FINISHED                                                                                                           
 => [internal] load .dockerignore                                                                                                     0.1s
 => => transferring context: 2B                                                                                                       0.0s
 => [internal] load build definition from Dockerfile                                                                                  0.1s
 => => transferring dockerfile: 130B                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                      2.7s
 => CACHED [1/2] FROM docker.io/library/alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad                0.1s
 => => resolve docker.io/library/alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad                       0.1s
 => ERROR [2/2] RUN apk add neofetch                                                                                                  0.3s
------
 > [2/2] RUN apk add neofetch:
#0 0.260 ERROR: Unable to lock database: Permission denied
#0 0.263 ERROR: Failed to open apk database: Permission denied
------
Dockerfile:2
--------------------
   1 |     FROM alpine
   2 | >>> RUN apk add neofetch
   3 |     
--------------------
error: failed to solve: process "/bin/sh -c apk add neofetch" did not complete successfully: exit code: 99
```

```console
# ausearch -m avc
time->Wed Oct 19 15:31:03 2022
type=AVC msg=audit(1666193463.724:567): avc:  denied  { write } for  pid=2031 comm="apk" name="lock" dev="overlay" ino=94946 scontext=system_u:system_r:container_t:s0:c23,c164 tcontext=unconfined_u:object_r:container_ro_file_t:s0 tclass=file permissive=0
```

File: container.fc [hunk 1]
```diff
@@ -66,6 +71,9 @@
 /var/lib/containerd/[^/]*/snapshots(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)
 /var/lib/containerd/[^/]*/sandboxes(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)
 
+/var/lib/buildkit(/.*)?	gen_context(system_u:object_r:container_var_lib_t,s0)
+/var/lib/buildkit/[^/]*/snapshots(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)
```
Comment (AkihiroSuda) [inline container.fc:75]: (I just copied the line from the containerd lines, but this seems wrong)

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
You might need something to match these lines?
/var/lib/containers/overlay(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)

/var/lib/buildkit/runc-overlay(/.*)?	gen_context(system_u:object_r:container_var_lib_t,s0)

Although I have no idea why you call this runc-overlay, seems overlay should be sufficient.

**Pull request merged**